treasuries/src/tests.rs

```diff
@@ -72,3 +74,222 @@ fn treasury_getter_works() {
 		)
 	});
 }
+#[rstest(burn, case(false), case(true))]
+fn swap_native_partial_works(burn: bool) {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(System::block_number() + 1); // this is needed to assert events
+		let beneficiary = AccountId::from(AccountKeyring::Alice);
+		let native_allowance: BalanceOf<TestRuntime> = 110_000_000;
+		let rate_float = 0.000_000_2;
+		let rate = Some(BalanceType::from_num(rate_float));
+		let cid = CommunityIdentifier::default();
+		let swap_option: SwapNativeOption<Balance, Moment> = SwapNativeOption {
+			cid,
+			native_allowance,
+			rate,
+			do_burn: burn,
+			valid_from: None,
+			valid_until: None,
+		};
```

Comment:
I don't see any tests, for the validity period, but I don't see that you check it yet either in the code. What is the plan here?

Reply:
I'm convinced that we will need these bounds in the next step and I want to avoid a state migration. may be premature optimization, but....

Reply:
okok